BootloaderCommonPkg/Library/ElfLib/Elf32.h

```diff
@@ -42,214 +35,218 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
  */
 
 #ifndef _SYS_ELF32_H_
-#define _SYS_ELF32_H_   1
+#define _SYS_ELF32_H_ 1
+
 
-#include "ElfCommon.h"
+/*
+ * ELF definitions common to all 32-bit architectures.
+ */
 
-//
-// ELF definitions common to all 32-bit architectures.
-//
-typedef UINT32       ELF32_ADDR;
-typedef UINT16       ELF32_HALF;
-typedef UINT32       ELF32_OFF;
-typedef INT32        ELF32_SWORD;
-typedef UINT32       ELF32_WORD;
-typedef UINT64       Elf32_LWORD;
+typedef UINT32  Elf32_Addr;
+typedef UINT16  Elf32_Half;
+typedef UINT32  Elf32_Off;
+typedef INT32    Elf32_Sword;
+typedef UINT32  Elf32_Word;
+typedef UINT64  Elf32_Lword;
 
-typedef ELF32_WORD   Elf32_HASHELT;
+typedef Elf32_Word  Elf32_Hashelt;
 
-// Non-standard class-dependent datatype used for abstraction.
-typedef ELF32_WORD   ELF32_SIZE;
-typedef ELF32_SWORD  ELF32_SSIZE;
+/* Non-standard class-dependent datatype used for abstraction. */
+typedef Elf32_Word  Elf32_Size;
+typedef Elf32_Sword  Elf32_Ssize;
+
+/*
+ * ELF header.
+ */
 
-//
-// ELF header.
-//
 typedef struct {
-  UINT8             E_Ident[EI_NIDENT];      // File identification.
-  ELF32_HALF        E_Type;                  // File type.
-  ELF32_HALF        E_Machine;               // Machine architecture.
-  ELF32_WORD        E_Version;               // ELF format version.
-  ELF32_ADDR        E_Entry;                 // Entry point.
-  ELF32_OFF         E_Phoff;                 // Program header file offset.
-  ELF32_OFF         E_Shoff;                 // Section header file offset.
-  ELF32_WORD        E_Flags;                 // Architecture-specific flags.
-  ELF32_HALF        E_Ehsize;                // Size of ELF header in bytes.
-  ELF32_HALF        E_Phentsize;             // Size of program header entry.
-  ELF32_HALF        E_Phnum;                 // Number of program header entries.
-  ELF32_HALF        E_Shentsize;             // Size of section header entry */
-  ELF32_HALF        E_Shnum;                 // Number of section header entries.
-  ELF32_HALF        E_Shstrndx;              // Section name strings section.
-} ELF32_EHDR;
-
-//
-// Section header.
-//
+  unsigned char  e_ident[EI_NIDENT];  /* File identification. */
+  Elf32_Half  e_type;    /* File type. */
+  Elf32_Half  e_machine;  /* Machine architecture. */
+  Elf32_Word  e_version;  /* ELF format version. */
+  Elf32_Addr  e_entry;  /* Entry point. */
+  Elf32_Off  e_phoff;  /* Program header file offset. */
+  Elf32_Off  e_shoff;  /* Section header file offset. */
+  Elf32_Word  e_flags;  /* Architecture-specific flags. */
+  Elf32_Half  e_ehsize;  /* Size of ELF header in bytes. */
+  Elf32_Half  e_phentsize;  /* Size of program header entry. */
+  Elf32_Half  e_phnum;  /* Number of program header entries. */
+  Elf32_Half  e_shentsize;  /* Size of section header entry. */
+  Elf32_Half  e_shnum;  /* Number of section header entries. */
+  Elf32_Half  e_shstrndx;  /* Section name strings section. */
```

Comment:
Both variables and comments are not following EDKII coding style, should we update them?

Reply:
Both Elf32.h and ElfCommon.h are from BaseTools/Source/C/GenFw/elf32.h and elf_common.h (FYI, no ELF library in EDKII). I am not sure we should update all of them. Any suggestions?

Reply:
If the header files comes from EDKII, we could keep it as-is with 1) update commit msg where they come from. 2) have same license header with EDKII (you did it already).

Reply:
Commit msg has the info - elf32.h/elf_common.h from BaseTools.

Reply:
Great. I missed that one.